src/tools/insertLinesTool.ts

```diff
@@ -1,72 +1,62 @@
-    type: 'object' as const,
-    properties: {
-      pageTitle: {
-        type: 'string',
-        description: 'Title of the page to modify',
-      },
-      targetLineText: {
-        type: 'string',
-        description:
-          'Text of the line after which to insert new content. If not found, content will be appended to the end.',
-      },
-      text: {
-        type: 'string',
-        description:
-          'Text to insert. If you want to insert multiple lines, use \\n for line breaks.',
-      },
+    {
+      title: z.string().describe('Title of the page to modify'),
+      targetLineText: z
+        .string()
+        .describe(
+          'Text of the line after which to insert new content. If not found, content will be appended to the end.'
+        ),
+      text: z
+        .string()
+        .describe(
+          'Text to insert. If you want to insert multiple lines, use \\n for line breaks.'
+        ),
     },
-    required: ['pageTitle', 'targetLineText', 'text'],
-  },
-  async execute(
-    { pageTitle, targetLineText, text }: InsertLinesArgs,
-    { projectName, cosenseOptions }: ToolContext
-  ): Promise<CallToolResult> {
-    const result = await patch(
-      projectName,
-      pageTitle,
-      (lines) => {
-        // find the index of the target line
-        let index = lines.findIndex((line) => line.text === targetLineText);
-        // if not found, append to the end
-        index = index < 0 ? lines.length : index;
-        // insert the text
-        const linesText = lines.map((line) => line.text);
-        return [
-          ...linesText.slice(0, index + 1),
-          ...text.split('\n'),
-          ...linesText.slice(index + 1),
-        ];
-      },
-      cosenseOptions
-    );
-
-    if (result.ok) {
-      return {
-        content: [
-          {
-            type: 'text',
-            text: `Successfully inserted lines.`,
-          },
-        ],
+    async ({ title, targetLineText, text }) => {
+      const cosenseOptions = {
+        sid: config.cosenseSid,
       };
-    } else {
-      throw unwrapErr(result);
+
+      const result = await patch(
+        config.projectName,
+        title,
+        (lines) => {
+          // find the index of the target line
+          let index = lines.findIndex((line) => line.text === targetLineText);
+          // if not found, append to the end
+          index = index < 0 ? lines.length : index;
+          // insert the text
+          const linesText = lines.map((line) => line.text);
+          return [
+            ...linesText.slice(0, index + 1),
+            ...text.split('\n'),
+            ...linesText.slice(index + 1),
+          ];
+        },
+        cosenseOptions
+      );
+
+      if (result.ok) {
+        return {
+          content: [
+            {
+              type: 'text',
+              text: `Successfully inserted lines.`,
+            },
+          ],
+        };
+      } else {
+        throw unwrapErr(result);
+      }
```

Comment:
Fixed! Changed to return content with `isError: true` instead of throwing the error, consistent with other tools like `getPageTool`.